### ISSUE TITLE
Edge detectors from sonnendemo

### ISSUE THREAD

**Issue by turion** [turion/rhine#80]
I implemented edge detectors in https://github.com/turion/sonnendemo. They should be part of the main library.

**Comment by ggreif** [comment]
@turion What about _hysteresis_? I.e. if the input wanders out of a (pre-)specified range, fire the rest of the computation.

**Comment by turion** [comment]
Sorry, I don't completely understand. Can you elaborate? In my understanding, hysteresis is some kind of memory/inertia. How is it related to what you mean? (Anyways, sounds like a separate issue to me)

**Comment by ggreif** [comment]
Yes it is stateful. I was thinking of flank detection, which is a (kind-of) building block of hysteresis. I should try to understand what your edge detection does precisely. Sorry for the noise.

**Comment by turion** [comment]
That proposal for edge detection is not very sophisticated. Surely, more complex approaches can be implemented in Rhine, e.g. measuring the energy of derivatives and so on. I'll leave that for future work.
